test/services/EtmpRegimeServiceTest.scala

```diff
@@ -101,50 +89,359 @@ class EtmpRegimeServiceTest extends BaseSpec {
+      val businessCustomerDetails = BusinessCustomerDetails("ACME Trading", Some("Corporate Body"), businessAddress , "1234567890",
+        "XE0001234567890", false, Some("XAAW00000123456"), Some("AARN1234567"), None, None)
+      val businessRegistrationDetails = BusinessRegistrationDetails(None, None, None, None, None, None, None, None, None)
+      val result = TestEtmpRegimeService.checkETMPApi(safeId, businessCustomerDetails, businessRegistrationDetails)
 
       await(result) shouldBe None
     }
 
     "fail to return a regimeRefNumber when the feature switch is disabled" in {
       FeatureSwitch.disable(AWRSFeatureSwitches.regimeCheck())
+      val businessAddress = BCAddress("1 LS House", "LS Way", Some("LS"), Some("Line 4"), Some("Postcode"), "GB")
+      val businessCustomerDetails = BusinessCustomerDetails("ACME Trading", Some("Corporate Body"), businessAddress , "1234567890",
+        "XE0001234567890", false, Some("XAAW00000123456"), Some("AARN1234567"), None, None)
+      val businessRegistrationDetails = BusinessRegistrationDetails(None, None, None, None, None, None, None, None, None)
+      val result = TestEtmpRegimeService.checkETMPApi(safeId, businessCustomerDetails, businessRegistrationDetails)
+
+      await(result) shouldBe None
+    }
+
+    "fails to return registration details when upserting enrolment fails" in {
+      FeatureSwitch.enable(AWRSFeatureSwitches.regimeCheck())
 
-      val result = TestEtmpRegimeService.checkETMPApi(safeId, regime)
+      when(mockAuthConnector.authorise[Option[AffinityGroup]](Matchers.any(), Matchers.any())(Matchers.any(), Matchers.any()))
+        .thenReturn(Future.successful(Some(AffinityGroup.Organisation)))
+      when(mockEtmpConnector.awrsRegime(Matchers.any())(Matchers.any()))
+        .thenReturn(Future.successful(HttpResponse(OK, Some(Json.parse(AwrsTestJson.getRegDetailsOrgWithRegime)))))
+      when(mockEnrolmentStore.upsertEnrolment(Matchers.any(), Matchers.any())(Matchers.any()))
+        .thenReturn(Future.failed(new RuntimeException("failure to return registration details")))
+
+      val businessAddress = BCAddress("1 LS House", "LS Way", Some("LS"), Some("Line 4"), Some("Postcode"), "GB")
+      val businessCustomerDetails = BusinessCustomerDetails("ACME Trading", Some("Corporate Body"), businessAddress , "1234567890",
+        "XE0001234567890", isAGroup = false, Some("XAAW00000123456"), Some("AARN1234567"), None, None)
+      val businessRegistrationDetails = BusinessRegistrationDetails(None, None, None, None, None, None, None, None, None)
+      val registrationDetails = EtmpRegistrationDetails(Some("ACME Trading"), "1234567890", "XE0001234567890", Some(false), "XAAW00000123456", Some("AARN1234567"), None, None)
+      val result = TestEtmpRegimeService.checkETMPApi(safeId, businessCustomerDetails, businessRegistrationDetails)
 
       await(result) shouldBe None
     }
   }
+
+  "handleDuplicateSubscription" should {
```

Comment:
GREY - Nitpick, but it feels like these tests are just testing getEtmpRegistrationDetails making the next set of tests partially redundant! Maybe these two functions didn't need to be split?